debezium-embedded/src/main/java/io/debezium/embedded/EmbeddedEngine.java

```diff
@@ -342,6 +342,81 @@ public boolean hasError() {
         }
     }
 
+    /**
+     * Interfaces with the maybeFlush and commitRecord APIs to allow
+     * the ChangeConsumer to specify when commits happen
+     */
+    public static interface RecordCommitter {
+        /**
+         * Marks a single record as processed, must be called for each
+         * record, and is threadsafe
+         * @param record the record to commit
+         */
+        void markProcessed(SourceRecord record) throws InterruptedException;
+
+        /**
+         * Marks a batch as finished, this may result in committing offsets/flushing
+         * data
+         *
+         * Should be called when a batch of records is finished being processed, but can
```

Comment:
I'm tripping over the "but can be called after..." part.

IIUC, this method is to be called by _implementors_ of `ChangeConsumer`, which in turn are invoked by the embedded engine itself. So only the EE will be in the position to invoke something "after `ChangeConsumer#handleBatch()` returns". So that second part of the sentence seems a bit contradictory?

Reply:
yeah... that isn't very good wording, basically it means that `markBatchProcessed` can be called asynchronously, i.e. after the `handleBatch` implementation returns control to the EmbededEnginge, will try and make that more clear... or just not specify it at all

Reply:
Thinking about that more... threadsafe sort of implies that, so will probably just simplify it